taca/analysis/analysis.py

```diff
@@ -130,8 +132,38 @@ def trigger_analysis(run_id):
                          "of {}. Please check the logfile and make sure to "
                          "start the analysis!".format(os.path.basename(run_id))))
 
+def find_samplesheet(run):
+    """Find the samplesheet for the given run
 
-def prepare_sample_sheet(run):
+    :param taca.illumina.Run run: Run to find the SampleSheet for
+    :returns str: Path to the SampleSheet
+    """
+    if run.run_type == 'MiSeq':
+        return os.path.join(run.run_dir, 'Data', 'Intensities', 'BaseCalls', 'SampleSheet.csv')
+    else:
+        current_year = '20' + run.id[0:2]
+        samplesheets_dir = os.path.join(CONFIG['analysis']['samplesheets_dir'],
+                                        current_year)
+        FC_ID = parsers.get_flowcell_id(run.run_dir)
+        return os.path.join(samplesheets_dir, '{}.csv'.format(FC_ID))
+
+
+def prepare_sample_sheet(run_dir, run_type, ss_origin):
+    """Prepares a non-X10 samplesheet
```

Comment:
@guillermo-carrasco the same here.... this and the next can be two specialisation of an abstract method.